### PR TITLE
feat/Appliance: Frontend Pulls Version Info from RelReg

### PR DESCRIPTION
**Pull request by Chickensoupwithrice**
Resolves [REL-290](https://linear.app/sourcegraph/issue/REL-290/frontend-ui-pulls-sourcegraph-versions-to-install-from-release) by making a single request to the release registry to get a list of versions

## Test plan 
manual tested

## Changelog

- feat(appliance): frontend pulls versions from relreg

### CODE REVIEW

**Comment by Chickensoupwithrice** [comment]
This feels like the wrong approach tbh, calling relregistry from clientside, when we have a release registry client in the backend is a bit silly. But then I'd have to expose the api endpoint on the backend that calls another api endpoint (the relreg)

Doing it this way means we need to pass another env var to the front end, which isn't the end of the world, but is annoying. 

i'm also still struggling to get a build of the maintaince ui running to test this, so i'm not actually sure it works, hence it's a draft

**Comment by Chickensoupwithrice** [comment]
Just remembered that I need to do the whole CORS dance 😭

**Review by craigfurman — COMMENTED** [review]
Yeah, this errors in my browser due to lack of CORS bypass.

Sadly we appear to have lost unified instructions on how to kick the tires on the appliance in development since we introduced the new UI.

You should be able to test this by: run the appliance backend, using instructions in https://github.com/sourcegraph/sourcegraph/blob/main/cmd/appliance/README.md.

In another terminal:

```
cd internal/appliance/frontend/maintenance
pnpm i
pnpm run dev
```

Assuming you get no build errors, navigate to localhost:8889 in your browser and you should see the maintenance UI.

**Comment by Chickensoupwithrice** [comment]
Thanks for the tips @craigfurman, I managed to get a test environment setup :) 

However it looks like CORS headers aren't being set on the releaseregistry, so I'm gonna make a PR there too

**Comment by Chickensoupwithrice** [comment]
Alright, finally got this tested, and working :) 

image

Requires changes to releaseregistry such that it also passes CORS headers, which I've made [here](https://github.com/sourcegraph/releaseregistry/pull/41)

File: internal/appliance/frontend/maintenance/src/Install.tsx [hunk 1]
```diff
@@ -1,14 +1,40 @@
...
 import { Button, FormControl, InputLabel, MenuItem, Paper, Select, Stack, Typography } from '@mui/material'
 
 import { changeStage } from './state'
 
 export const Install: React.FC = () => {
-    const [version, setVersion] = useState<string>('5.5.0')
+    const [version, setVersion] = useState<string[]>([])
+    const [selectedVersion, setSelectedVersion] = useState<string>('')
+
+    useEffect(() => {
```
Comment (craigfurman) [inline internal/appliance/frontend/maintenance/src/Install.tsx:11]: I have zero react experience so take this with a tablespoon of salt, but this view file doesn't _feel_ like quite the right place for this function. Maybe we put it in a ts file and import it? Totally non-blocking note in any case.

**Review by craigfurman — APPROVED** [review]
Nice work! 🚀 

I haven't reproduced the testing since (if I understand correctly) I'd need to populate some data in a local relreg until the CORS headers PR lands.

File: internal/appliance/frontend/maintenance/src/Install.tsx (hunk 1 above)
Comment (DaedalusG) [reply internal/appliance/frontend/maintenance/src/Install.tsx:11]: This is pretty typical in react with hooks in my experience

**Review by DaedalusG — APPROVED** [review]

File: internal/appliance/frontend/maintenance/src/Install.tsx [hunk 2]
```diff
@@ -1,14 +1,40 @@
...
+                    headers: {
+                        Authorization: `Bearer token`,
+                        'Content-Type': 'application/json',
+                    },
+                    mode: 'cors',
+                })
+                const data = await response.json()
+                setVersion(data)
+                if (data.length > 0) {
+                    const publicVersions = data.filter(item => item.public).map(item => item.version)
```
Comment (DaedalusG) [inline internal/appliance/frontend/maintenance/src/Install.tsx:24]: The critical bit here 👍🏼

**Pull request merged**